File: src/ts/read-numbers.ts
import fs from "fs";

function readNumbers() {
  const filepath = process.argv[2];
  if (!filepath) return;

  const lines = fs.readFileSync(filepath).toString().split("\n");
  lines.forEach((line) => {
    const print = parseInt(line);

    if (isNaN(print)) {
      console.log("Line not a number");
    } else {
      console.log(line);
    }
  });
}

readNumbers();

File: src/rust/read_numbers.rs
pub fn read_numbers() {
    let file_name = std::env::args().nth(1).expect("the file name to passed in");

    let file = std::fs::read_to_string(file_name).expect("unable to read the file to string");

    file.lines().for_each(|line| {
        if let Ok(value) = line.parse::<usize>() {
            println!("{}", value);
        } else {
            println!("Line is not a number");
        }
        // match line.parse::<usize>() {
        //     Ok(line) => println!("{}", line),
        //     Err(_) => println!("Line is not a number"),
        // };
    });
}
